### PR TITLE
Migrated project to typescript

### PR DESCRIPTION
**Pull request by terret**
**updated**:
- index.js -> src/index.ts
- circleci/config.yml

**added:** 
- babelrc;
- tsconfig;
- eslintignore;

### CODE REVIEW

File: .babelrc.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  plugins: ['@babel/plugin-proposal-object-rest-spread'],
```
Comment (vladgolubev) [inline .babelrc.js]:
```suggestion
```

Actually, this line can be removed

File: package.json [hunk 2]
```diff
@@ -2,65 +2,78 @@
...
   "author": {
     "name": "Vlad Holubiev",
     "email": "golubev.vld@gmail.com",
     "url": "vladholubiev.com"
   },
-  "engines": {
-    "node": ">=6"
-  },
+  "files": [
+    "src/index.ts"
```
Comment (vladgolubev) [inline package.json]:
```suggestion
    "lib"
```

`files` is what's getting published with the npm package

File: package.json [hunk 3]
```diff
@@ -2,65 +2,78 @@
...
-    "*.{ts,js}": [
-      "eslint --fix",
-      "git add"
-    ],
-    "*.{html,json,md,yml}": [
-      "prettier --write --ignore-path=./.eslintignore",
-      "git add"
-    ]
+  "engines": {
+    "node": ">=6"
```
Comment (vladgolubev) [inline package.json]: Since you've set target Node 12 in the `.babelrc.js`, the same version should be here as well

File: package.json [hunk 4]
```diff
@@ -2,65 +2,78 @@
...
     "string-length": "3.1.0"
   },
   "devDependencies": {
+    "@babel/cli": "^7.8.0",
+    "@babel/core": "^7.8.0",
+    "@babel/preset-env": "^7.8.2",
+    "@babel/preset-typescript": "^7.8.0",
     "@shelf/eslint-config": "0.13.0",
+    "@types/jest": "^24.0.25",
+    "@types/node": "^13.1.6",
```
Comment (vladgolubev) [inline package.json]:
```suggestion
    "@types/node": "12",
```

Since this is the target version in your `.babelrc.js`

File: package.json [hunk 5]
```diff
@@ -2,65 +2,78 @@
...
-    "utf",
-    "utf8",
-    "substring"
-  ],
   "dependencies": {
     "runes": "0.4.3",
     "string-length": "3.1.0"
   },
   "devDependencies": {
+    "@babel/cli": "^7.8.0",
```
Comment (vladgolubev) [inline package.json]: Please install packages with the `--exact` flag, like `yarn add @babel/cli --exact`. See that `dependencies` don't have `^` symbol in their version names

File: wallaby.config.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
 module.exports = () => {
   return {
     testFramework: 'jest',
-    files: ['package.json', 'index.js', '!index.test.js'],
-    tests: ['index.test.js'],
+    files: ['package.json', 'index.ts', '!index.test.ts'],
```
Comment (vladgolubev) [inline wallaby.config.js]: Since you moved everything to the `src` folder, make sure paths are up-to-date here as well

**Review by vladholubiev — APPROVED** [review]

**Comment by vladgolubev** [comment]
@terret test

**Comment by vladgolubev** [comment]
@terret test

**Review by KnupMan — COMMENTED** [review]
:+1:

**Pull request merged**